### PR TITLE
[DOCS] Add docs for new Lucene's filters for Japanese text.

### PR DESCRIPTION
**Pull request by daixque**
This PR adds a documentation for new Lucene's filters for Japanese text under analysis-kuromoji plugin.

New filters are:
 - hiragana_uppercase
 - katakana_uppercase

These filters are introduced to Lucene 9.11 and also it's available on Elasticsearch from 8.15.

This is related to:
* #106553

### CODE REVIEW

File: docs/plugins/analysis-kuromoji.asciidoc [hunk 1]
```diff
@@ -624,3 +624,123 @@ Which results in:
   } ]
 }
 --------------------------------------------------
+
+[[analysis-kuromoji-hiragana-uppercase]]
+==== `hiragana_uppercase` token filter
+
+The `hiragana_uppercase` token filter normalizes small letters (捨て仮名) in hiragana into normal letters.
```
Comment (leemthompo) [inline docs/plugins/analysis-kuromoji.asciidoc:631]: Is "normal letters" accepted phrasing?

Maybe "The `hiragana_uppercase` token filter normalizes small Hiragana letters (捨て仮名) into full-size Hiragana letters?

File: docs/plugins/analysis-kuromoji.asciidoc [hunk 2]
```diff
@@ -624,3 +624,123 @@ Which results in:
...
+      "position": 2
+    }
+  ]
+}
+--------------------------------------------------
+
+[[analysis-kuromoji-katakana-uppercase]]
+==== `katakana_uppercase` token filter
+
+The `katakana_uppercase` token filter normalizes small letters (捨て仮名) in katakana into normal letters.
```
Comment (leemthompo) [inline docs/plugins/analysis-kuromoji.asciidoc:698]: Same question as above.

**Review by leemthompo — APPROVED** [review]
LGTM. I tested that this works as documented. I have one minor non-blocking question about wording.

 Thanks for opening PR and adding to the docs 🏅

File: docs/plugins/analysis-kuromoji.asciidoc (hunk 1 above)
Comment (daixque) [reply docs/plugins/analysis-kuromoji.asciidoc:631]: Good point, maybe `standard` (or `regular`) would be better. The word "Full-size" sounds like "full-width" (multi-byte), which is not the case here. Let me change "normal" to "standard".

**Comment by leemthompo** [comment]
🚢

**Pull request merged**